File: stacks/Stack.java
public class Stack<T> {
    private Node first;
    private int num;

    private class Node {
        T value;
        Node next;
    }

    public void push(T value) {
        Node newNode = new Node();
        newNode.value = value;
        newNode.next = first;
        first = newNode;
        num++;
    }

    public Node pop() {
        Node topNode = first;
        first = topNode.next;
        num--;
        return topNode;
    }

    public int size() {
        return num;
    }

    public boolean isEmpty() {
        return num == 0;
    }

    public static void main(String[] args) {
        Stack theStack = new Stack<Integer>();
        theStack.push(3);
        theStack.push(2);
        theStack.push(1);
        System.out.println("Size: " + theStack.size());
        System.out.println("isEmpty: " + theStack.isEmpty());
        System.out.println("Popped value: " + theStack.pop().value);
        System.out.println("Size: " + theStack.size());
        System.out.println("isEmpty: " + theStack.isEmpty());
        System.out.println("Popped value: " + theStack.pop().value);
        System.out.println("Size: " + theStack.size());
        System.out.println("isEmpty: " + theStack.isEmpty());
        System.out.println("Popped value: " + theStack.pop().value);
        System.out.println("Size: " + theStack.size());
        System.out.println("isEmpty: " + theStack.isEmpty());
        // System.out.println("Popped value: " + theStack.pop().value); // NullPointerException
    }

}

File: stacks/stack.py
class Stack:

    first = None
    num = 0

    def __init__(self):
        self.first = None
        self.num = 0

    def push(self, value):
        newNode = Node(value, self.first)
        self.first = newNode
        self.num += 1

    def pop(self):
        topNode = self.first
        self.first = topNode.next
        self.num -= 1
        return topNode

    def size(self):
        return self.num

    def isEmpty(self):
        return self.num == 0

class Node:
    def __init__(self, value, nextNode):
        self.value = value
        self.next = nextNode

myStack = Stack()
myStack.push(3)
myStack.push(2)
myStack.push(1)
print("Size: " + str(myStack.size()))
print("isEmpty:" + str(myStack.isEmpty()))
print("")
#
print("Value: " + str(myStack.pop().value))
print("Size: " + str(myStack.size()))
print("isEmpty:" + str(myStack.isEmpty()))
print("")
#
print("Value: " + str(myStack.pop().value))
print("Size: " + str(myStack.size()))
print("isEmpty:" + str(myStack.isEmpty()))
print("")
#
print("Value: " + str(myStack.pop().value))
print("Size: " + str(myStack.size()))
print("isEmpty:" + str(myStack.isEmpty()))
